src/one-time-token/controller.ts

```diff
@@ -0,0 +1,39 @@
+import Controller from '@curveball/controller';
+import { Context } from '@curveball/core';
+import * as privilegeService from '../privilege/service';
+import { Forbidden, NotFound } from '@curveball/http-errors';
+import * as userService from '../user/service';
+import { User } from '../user/types';
+import { createToken } from '../reset-password/service';
+
+class OneTimeTokenController extends Controller {
+
+  async post(ctx: Context<any>) {
+
+    if (!await privilegeService.hasPrivilege(ctx, 'admin')) {
+      throw new Forbidden('Only users with the "admin" privilege can create new users');
+    }
+
+    let user: User;
+    try {
+      user = await userService.findByIdentity(ctx.state.user.identity);
+    } catch (err) {
+      if (err instanceof NotFound) {
+        ctx.redirect(303, '/reset-password?error=Account+not+found.+Please+try+again');
+        return;
+      } else {
+        throw err;
+      }
+    }
+
+    const token = await createToken(user);
+    const url = process.env.PUBLIC_URI + 'reset-password/token/' + token;
```

Comment:
returning the URL with the token sounds like a good idea to me, but a good way to return a url is with a hal link.